admission/hooks/mutate_pod.go

```diff
@@ -0,0 +1,101 @@
+package hooks
+
+import (
+	"context"
+	"crypto/sha1"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strconv"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+// +kubebuilder:webhook:verbs=create;update,path=/mutate-pod,mutating=true,failurePolicy=fail,groups="",resources=pods,versions=v1,name=mpod.kb.io
+
+type podMutator struct {
+	client  client.Client
+	decoder *admission.Decoder
+}
+
+// NewPodMutator creates a webhook handler for Pod.
+func NewPodMutator(c client.Client, dec *admission.Decoder) http.Handler {
+	return &webhook.Admission{Handler: &podMutator{c, dec}}
+}
+
+func (m *podMutator) Handle(ctx context.Context, req admission.Request) admission.Response {
+	po := &corev1.Pod{}
+	err := m.decoder.Decode(req, po)
+	if err != nil {
+		return admission.Errored(http.StatusBadRequest, err)
+	}
+
+	poPatched := po.DeepCopy()
+	for i, co := range po.Spec.Containers {
+		if !m.isMountedTmp(&co) {
+			volumeName := m.generateVolumeName(co.Name, po.Spec.Volumes)
+			m.appendEmptyDir(volumeName, poPatched)
+			poPatched.Spec.Containers[i].VolumeMounts = append(poPatched.Spec.Containers[i].VolumeMounts,
+				corev1.VolumeMount{Name: volumeName, MountPath: "/tmp"})
+		}
+	}
+	for i, co := range po.Spec.InitContainers {
+		if !m.isMountedTmp(&co) {
+			volumeName := m.generateVolumeName(co.Name, po.Spec.Volumes)
+			m.appendEmptyDir(volumeName, poPatched)
+			poPatched.Spec.InitContainers[i].VolumeMounts = append(poPatched.Spec.InitContainers[i].VolumeMounts,
+				corev1.VolumeMount{Name: volumeName, MountPath: "/tmp"})
+		}
+	}
+
+	marshaled, err := json.Marshal(poPatched)
+	if err != nil {
+		return admission.Errored(http.StatusInternalServerError, err)
+	}
+	return admission.PatchResponseFromRaw(req.Object.Raw, marshaled)
+}
+
+func (m *podMutator) isMountedTmp(co *corev1.Container) bool {
+	for _, mount := range co.VolumeMounts {
+		if mount.MountPath == "/tmp" || strings.HasPrefix(mount.MountPath, "/tmp/") {
+			return true
+		}
+	}
+	return false
+}
+
+func (m *podMutator) hashString(name string) string {
+	h := sha1.New()
+	h.Write([]byte(name))
+	bs := h.Sum(nil)
+	return fmt.Sprintf("%x", bs)
```

Comment:
This can be written simply as:
```go
        hex.EncodeToString(sha1.Sum([]byte(name)))
```

Reply:
Thanks. I fixed it.
And the hash string was too long, so I cut it to 16 characters.

Reply:
Would you please describe why it is limited to 16 characters?
And if there are any clear reasons, please add comments on the code.

Reply:
The limitation(16 characters) has no meaning.

I checked the hash suffix generated by Kubernetes, it is 5 characters for Pod.
So, I will change the length to 5 characters just like a Pod.

Reply:
If there is no reason, I'd like to keep the hash string as is (40 bytes?).

Reply:
OK. I will remove the limitation. The generated volume name is as follows. (`tmp-`+55 charactoers)
```
tmp-7562756e747530da39a3ee5e6b4b0d3255bfef95601890afd80709
```

Reply:
That's strange...  SHA1 is 20 bytes, which will be converted to a 40 byte hex string.

Reply:
Got it.  Your usage of `sha1.New().Sum(...)` was wrong.  I fixed it.